src/gpl/src/initialPlace.cpp

```diff
@@ -41,6 +43,40 @@
 #include "graphics.h"
 
 #include "utl/Logger.h"
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include "cusparse.h"
+#include "cusolverDn.h"
+#include "cusolverSp.h"
+
+#define CUDA_ERROR(value) do { \
+   cudaError_t m_cudaStat = value; \
+   if(m_cudaStat != cudaSuccess){ \
+       fprintf(stderr, "Error %s at line %d in file %s\n", cudaGetErrorString(m_cudaStat), __LINE__, __FILE__); \
+       exit(-1); \
```

Comment:
All messages should go through the logger.  These should be error() calls which will generate an exception.  exit() is never used as a form of error handling.